### PR TITLE
Python parser refactor and unused tokens clean up

### PR DESCRIPTION
**Pull request by o01eg**
Fully cleans not implemented tokens.

Follows up #3860

### CODE REVIEW

File: parse/IntValueRefParser.cpp [hunk 1]
```diff
@@ -33,7 +33,6 @@ parse::detail::simple_int_parser_rules::simple_int_parser_rules(const parse::lex
         |   tok.FleetID_
         |   tok.PlanetID_
         |   tok.SystemID_
-        |   tok.ContainerID_
```
Comment (geoffthemedio) [inline parse/IntValueRefParser.cpp]: What is your definition of "unused"?

https://github.com/freeorion/freeorion/blob/master/universe/ValueRefs.cpp#L1051

File: parse/ArithmeticRules.cpp [hunk 2]
```diff
@@ -53,7 +53,6 @@ namespace parse::detail {
                         |   tok.NoOp_   [ _c = ValueRef::OpType::NOOP ] // pass-through does nothing, for debug purposes
                         |   tok.abs_    [ _c = ValueRef::OpType::ABS ]
                         |   tok.round_  [ _c = ValueRef::OpType::ROUND_NEAREST ]
-                        |   tok.ceil_   [ _c = ValueRef::OpType::ROUND_UP ]
```
Comment (geoffthemedio) [inline parse/ArithmeticRules.cpp]: this is removing functionality that ValueRefs support... https://github.com/freeorion/freeorion/blob/master/universe/ValueRefs.cpp#L3372

File: parse/IntValueRefParser.cpp (hunk 1 above)
Comment (o01eg) [reply parse/IntValueRefParser.cpp]: Not mentioned in FOCS files. Some of them even not used in parser code.

File: parse/ArithmeticRules.cpp (hunk 2 above)
Comment (o01eg) [reply parse/ArithmeticRules.cpp]: They will remain accessible from Python Parser.

Comment (geoffthemedio) [reply parse/ArithmeticRules.cpp]: I'd rather not remove functionality from the Boost parser until the migration to Python is complete.

File: parse/IntValueRefParser.cpp (hunk 1 above)
Comment (geoffthemedio) [reply parse/IntValueRefParser.cpp]: The ones that aren't even reference in parser code are fine to remove.

**Pull request merged**